### DISCUSSION TITLE
ADC/TDC "order" conventions

### DISCUSSION THREAD

**Discussion by baltzell** [JeffersonLab/clas12-offline-software#965] [Ideas]
**Currently standard, valid values for `order` are 0-9.  Correct?  We don't really use the space above 4, right?**

To unify treatment/interpretation of this `order` variable in ADC/TDC banks to mark hits, we need to define a convention for at least:

* Background hits from beam overlay
* Hits rejected by background
* Whatever other mark someone wants to apply

Here's one proposal:
* Background hits take `order+10`
* Background-rejected hits takes `order+100`
* User-defined marks are `order+20` if the hit should be used in tracking, else `order+120`  (8/9 would be off limits, see below)

_Note, the `order` variable is 1 byte, so to keep it human readable in java without having to manipulate it, it must be in the range [-128,127]._  Human readability isn't necessarily a requirement, but would be nice.

**Comment by baltzell** [comment]
And here's Gagik's minimial proposal for an ADC/TDC-provider class, which can be amended with conveniences after finalizing a convention:

https://gist.github.com/baltzell/e448e0f6c1e26ce5bbe9db543dd9b494

**Comment by baltzell** [comment]
Will any calibration suites need updating?

**Reply by hauenst** [reply]
BAND calibration is using orders but only 0-3, so I dont think we need any updating

**Reply by baltzell** [reply]
Any calibration suite that really uses order at all would need updating to work on data with marked ADC/TDC hits.  So far, marking hits is for background merging or other custom purposes, not standard calibrations.
